sklearn/decomposition/tests/test_nmf.py

```diff
@@ -560,7 +567,7 @@ def test_nmf_float32_float64_consistency(solver, regularization):
     nmf64 = NMF(solver=solver, regularization=regularization, random_state=0)
     W64 = nmf64.fit_transform(X)
 
-    assert_allclose(W32, W64, rtol=1e-6, atol=1e-5)
+    assert_allclose(W32, W64, rtol=1e-5, atol=1e-4)
```

Comment:
"nndsvdar" adds some randomness to the init, which is probably why you had to decrease the precision here. I would prefer keeping previous precision, and use "nndsvda" instead, which should work fine since it it not random.